addons/udes_stock/models/abstract_stock_model.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+class AbstractStockModel(models.AbstractModel):
+    _name = 'abstract.stock.model'
+
+    # Whether the model can create or not
+    CREATE = False
+    # The default domain to search with a str
+    STR_DOMAIN = ['name']
+
+    def get_or_create(self, identifier, create=False):
+        """ Get or create in model, for create alos need CREATE to be True
+        :args:
+            - identifier: str or int
+                The identifier to search with
+        :kwargs:
+            - create: Boolean
+                If true, and CREATE is true, then a new object is created
+        :returns:
+            The model queried
+        """
+        model = self._name
+        model_type = model.split('.')[-1]
```

Comment:
Pick a better name- model_type is not clear.

Reply:
Is this one because model is to abstract? Do we want a user friendly name for the model?

Reply:
So it's the object name, i.e package, group, product etc. Suppose we could just use object name - it's for the error messages.

Reply:
OK, we can improve it later.